packages/patrol_cli/lib/src/runner/patrol_command_runner.dart

```diff
@@ -73,19 +73,35 @@ class PatrolCommandRunner extends CompletionCommandRunner<int> {
           'patrol',
           'Tool for running Flutter-native UI tests with superpowers',
         ) {
```

Comment:
I see that all the commands depend on `testFinder`, `dartDefinesReader`, `pubspecReader`, `androidTestBackend`, `iosTestBackend`, and `logger`. I'm thinking whether there's some smart way to bundle it all together and not pass it explicitly to each and every command, but without simply wrapping it in a class, but somehow that would make sense

Reply:
I'm also wondering about it quite often, but so far I haven't found a solution that would (1) reduce duplication and (2) be flexible and testable.

`flutter_tools` has [AppContext](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/context.dart) for this coupled with Zones, but I'm sure it's an overkill for `patrol_cli`.